internal/store/certificatesigningrequest_test.go

```diff
@@ -29,13 +29,13 @@ import (
 func TestCsrStore(t *testing.T) {
 	const metadata = `
 		# HELP kube_certificatesigningrequest_labels [STABLE] Kubernetes labels converted to Prometheus labels.
-		# TYPE kube_certificatesigningrequest_labels gauge
+		# TYPE kube_certificatesigningrequest_labels 0
```

Comment:
This seems unusual to me. Is this supported in prometheus?

Reply:
I understand that you're doing this only in the test. I think it would be helpful to have a conversion function as well and keep the metadata enduser friendly (e.g. I copy and paste something into my testbench).

Reply:
+1, but AFAICT we can only deduce the `content-type` once the request is made (`ServeHTTP`), which is where `SanitizeHeaders` is being invoked from currently, for this reason.

Moving the call further down in the stack, say, to where the headers are being generated wouldn't provide us enough information (`content-type`) to "sanitize" the headers.

To circumvent this, the current workflow is:
* append the `typeN` representation of the `metric.type` to the `# TYPE` header metadata,
* once a request is made, deduce if it was `expfmt.ProtoType`-based,
* * if so, `s/(infoN|statesetN)/gaugeN`
* finally, `s/xN/x` (convert `typeN` back to `metric.type`)

Reply:
That being said, the whole `s/gauge/0` etc., in the tests seems odd indeed, but I'm not quite sure if there's a better way around this.

Reply:
I just realized that we really don't need to search for anything, as opposed to my initial [train of thought](https://github.com/kubernetes/kube-state-metrics/pull/2270#issuecomment-1857791232). Let me patch this again.